### PR TITLE
[RHCLOUD-19793] Source handler tests update3

### PR DESCRIPTION
**Pull request by petracihalova**
Tests update for source handlers - part III. 
(part I. #369, part II. #370)

- main goal is to add/update tests to better tenancy testing
- each commit contains one updated or added test (or new helper function)
- changes for other functions in source handler will be part of next PR

handlers covered:
 - SourceEdit()
 - SourceDelete()
 - SourceCheckAvailability()
 - SourcesRhcConnectionList()

**JIRA:** [RHCLOUD-19793](https://issues.redhat.com/browse/RHCLOUD-19793)

### CODE REVIEW

**Comment by petracihalova** [comment]
PR will be rebased after merge #370

**Comment by petracihalova** [comment]
rebased and ready for review

File: source_handlers_test.go [hunk 1]
```diff
@@ -1485,7 +1581,68 @@ func TestSourcesGetRelatedRhcConnectionsTest(t *testing.T) {
 	}
 }
 
-func TestSourcesGetRelatedRhcConnectionsTestBadRequestNotFound(t *testing.T) {
+// TestSourcesGetRelatedRhcConnectionsEmptyList tests that you get empty list
+// for source without rhd-connections
```
Comment (MikelAlejoBR) [inline source_handlers_test.go]:
```diff
- rhd-connections
+ rhc-connections
```

**Review by MikelAlejoBR — APPROVED** [review]
Looks good to me!

File: source_handlers_test.go (hunk 1 above)
Comment (petracihalova) [reply source_handlers_test.go]: nice catch 👍🏻

**Comment by petracihalova** [comment]
rebased, comment solved, ready for review

**Review by MikelAlejoBR — APPROVED** [review]

**Review by lpichler — APPROVED** [review]

**Pull request merged**